pandas/core/algorithms.py

```diff
@@ -1105,6 +1105,22 @@ def _get_score(at):
         return result
 
 
+def check_percentile(q: Union[float, Iterable[float]]) -> np.ndarray:
+    """
+    Validate percentiles (used by describe and quantile).
+    """
+
+    msg = "percentiles should all be in the interval [0, 1]. " "Try {0} instead."
+    q = np.asarray(q)
```

Comment:
You can just create a new variable here `q_arr` instead of reusing the argument; should help with some of the typing errors